plugin.js

```diff
@@ -1,36 +1,70 @@
-      );
-    },
-    icons: 'bylawlist', // Bylaw List icon
-    init: function( editor ) {
-        editor.addCommand( 'insertBylawlist', {
-            exec: function( editor ) {
-              if ('li' == editor.getSelection().getStartElement().getName()) {  
-                if ('OL' == editor.getSelection().getStartElement().$.parentNode.nodeName) {
-                  var bylaw = editor.getSelection().getStartElement().$.parentNode;
-                  if(bylaw.className.indexOf("bylawlist") < 0){
-                     bylaw.className += "bylawlist";
-                  } 
-                  else {
-                    bylaw.className = "";
-                  }
-                  
-                }              
-              }
+CKEDITOR.plugins.add('bylawlist', {
+  onLoad: function () {
+    CKEDITOR.addCss(
+      'ol.bylawlist{list-style-type:none;list-style-position:inside;padding-left:1.5em}ol.bylawlist>li>ol,ol.bylawlist>li>ol>li>ol{list-style-type:inherit}ol.bylawlist li{counter-increment:bylawlist-counter}ol.bylawlist li:first-child{counter-reset:bylawlist-counter}ol.bylawlist>li:before{content:"(" counter(bylawlist-counter,decimal) ") "}ol.bylawlist>li>ol>li:before{content:"(" counter(bylawlist-counter,lower-alpha) ") "}ol.bylawlist>li>ol>li>ol>li:before{content:"(" counter(bylawlist-counter,lower-roman) ") "}ol.bylawlist>li>ol>li>ol>li>ol{list-style-type:decimal}ol.bylawlist>li>ol>li>ol>li>ol>li:before{content:none}'
+    );
+  },
+  icons: 'bylawlist', // Bylaw List icon
+  init: function (editor) {
+    editor.addCommand('insertBylawlist', {
+      exec: function (editor) {
+        if ('li' == editor.getSelection().getStartElement().getName()) {
+          if ('OL' == editor.getSelection().getStartElement().jQuery.parentNode.nodeName) {
+            var bylaw = editor.getSelection().getStartElement().jQuery.parentNode;
+
+            if (bylaw.className.indexOf("bylawlist") < 0) {
+              bylaw.className += "bylawlist";
+            } else {
+              bylaw.className = "";
             }
-       });
-        editor.ui.addButton( 'Bylawlist', {
-            label: 'Add Bylawlist',
-            command: 'insertBylawlist',
-            toolbar: 'insert'
+          }
+        }
+      }
+    });
+
+    editor.ui.addButton('Bylawlist', {
+      label: 'Add Bylawlist',
+      command: 'insertBylawlist',
+      toolbar: 'insert'
+    });
+
+    var format = {
+      'element': 'Bylawlist'
+    };
+
+    var style = new CKEDITOR.style(format);
+
+    editor.attachStyleStateChange(style, function (state) {
+      !editor.readOnly && editor.getCommand('insertBylawlist').setState(state);
+    });
+    editor.on('contentDom', function () {
```

Comment:
This is where most of the complexity was. Since the ckeditor uses an iframe / custom document element we had to jump into that document from the editor to attach the observers. This is what makes the numbers change when a user sets the starting value. This only works in browsers that support Mutation observers but Vennila said that's fine since all their authors are told to use Chrome.